src/psyclone/domain/lfric/kernel/field_arg.py

```diff
@@ -0,0 +1,177 @@
+        this information is captured in an fparser2 tree.
+
+        :param psyir: fparser2 tree containing the PSyIR for a field \
+            argument.
+        :type psyir: :py:class:`fparser.two.Fortran2003.Part_Ref`
+
+        :returns: an instance of FieldArg.
+        :rtype: :py:class:`psyclone.domain.lfric.kernel.FieldArg`
+
+        '''
+        FieldArg.check_psyir(psyir, nargs=4)
+        datatype = psyir.children[1].children[1].tostr()
+        access = psyir.children[1].children[2].tostr()
+        function_space = psyir.children[1].children[3].tostr()
+        return FieldArg(datatype, access, function_space)
+
+    @classmethod
+    def create_from_fortran_string(cls, fortran_string):
+        '''Create an instance of this class from a Fortran string.
+
+        :param str fortran_string: a string containing the metadata in \
+            Fortran.
+
+        :returns: an instance of cls.
+        :rtype: :py:class:`psyclone.domain.lfric.kernel.cls`
+
+        '''
+        part_ref = cls.create_psyir(fortran_string)
+        return cls.create_from_psyir(part_ref)
+
+    def fortran_string(self):
+        '''
+        :returns: the metadata represented by this class as a \
+            Fortran string.
+        :rtype: str
+
+        raises ValueError: if one or more of the datatype, access or \
+            function_space values have not been set.
+
+        '''
+        if not (self.datatype and self.access and self.function_space):
+            raise ValueError(
+                f"Values for datatype, access and function_space must be "
+                f"provided before calling the fortran_string method, but "
+                f"found '{self.datatype}', '{self.access}' and "
+                f"'{self.function_space}'.")
+
+        return (f"arg_type({self.form}, {self.datatype}, {self.access}, "
+                f"{self.function_space})")
+
+    @staticmethod
+    def check_datatype(value):
+        '''
+        :param str value: set the datatype to the \
+            specified value.
+
+        :raises ValueError: if the provided value is not a valid \
+            datatype descriptor.
+
+        '''
+        const = LFRicConstants()
+        if not value or value.lower() not in const.VALID_FIELD_DATA_TYPES:
+            raise ValueError(
+                f"The second metadata entry for an argument should "
+                f"be a recognised datatype descriptor (one of "
+                f"{const.VALID_FIELD_DATA_TYPES}), but found '{value}'.")
+
+    @staticmethod
+    def check_access(value):
+        '''
+        :param str value: set the access descriptor to the \
+            specified value.
+
+        :raises ValueError: if the provided value is not a valid \
+            access type.
+
+        '''
+        const = LFRicConstants()
+        if not value or value.lower() not in const.VALID_FIELD_ACCESS_TYPES:
+            raise ValueError(
+                f"The third metadata entry for an argument should "
```

Comment:
Strictly speaking, I'm not sure that this method should know that the access value should be the third metadata entry but it's a small point.

Reply:
Ignoring you then :-)